recipe/build.sh

```diff
@@ -12,29 +13,43 @@ else
 fi
 
 if [[ "$mpi" == "nompi" ]]; then
- VERSION=ssmp
+ CP2K_USE_MPI=OFF
 else
- VERSION=psmp
+ CP2K_USE_MPI=ON
 fi
-
-# make
-cp ${RECIPE_DIR}/${ARCH}.${VERSION} arch/${ARCH}.${VERSION}
-make -j${CPU_COUNT} ARCH=${ARCH} VERSION=${VERSION}
-
+mkdir build
+pushd build
+
+cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DCMAKE_FIND_FRAMEWORK=NEVER \
+      -DCMAKE_FIND_APPBUNDLE=NEVER \
+      -DPython3_EXECUTABLE="$PYTHON" \
+      -DCP2K_BUILD_DBCSR=ON \
+      -DCP2K_USE_MPI=$CP2K_USE_MPI \
+      -DCP2K_BLAS_VENDOR=OpenBLAS \
+      -DCP2K_SCALAPACK_VENDOR=GENERIC \
+      ${CMAKE_ARGS} \
+      ..
+
+cmake --build . --config Release
+
+make install
 # run regression tests
 if [[ "$mpi" == "nompi" ]]; then
-  make ARCH=${ARCH} VERSION=${VERSION} TESTOPTS="--ompthreads 2" test
+  make TESTOPTS="--ompthreads 2" test
```

Comment:
If the full test suites takes too long then you could also try the new `--smoketest` option.

Reply:
Also, when building with CMake then the test-script has to be called directly:

```
export CP2K_DATA_DIR=./data
./tools/regtesting/do_regtest.py ${ARCH} ${VERSION}
```

It expects to find the binary at `exe/${ARCH}/cp2k.${VERSION}`.